src/machines/signal_external_state_machine.rs

```diff
@@ -68,7 +68,7 @@ pub(super) fn new_external_signal(
         shared_state: SharedState { seq },
     };
     OnEventWrapper::on_event_mut(&mut s, SignalExternalMachineEvents::Schedule)
```

Comment:
Don't know where to put this comment, when cancelling the signal we might want to resolve the signal with failure.